ops/model.py

```diff
@@ -111,6 +112,11 @@ def name(self) -> str:
         """
         return self._backend.model_name
 
+    @property
+    def goal(self) -> 'GoalState':
+        """Return :class: GoalState object representing the goal-state of the Juju model."""
+        return self._cache.get(GoalState)
```

Comment:
unless I'm misunderstanding something, you're caching the `GoalState` object which is just a proxy, but not caching the actual call to the backend, meaning that every `self.model.goal.num_units` will fire a `goal-state`, right?